File: src/giada/swing/MnR/DefaultSliderModelAndRenderer.java
package giada.swing.MnR;

import def.dom.HTMLElement;
import giada.swing.JSlider;

/**
 * The default implementation of the AbstractSliderModelAndRenderer
 *
 * @author gianpiero.diblasi
 * @param <T> The type
 */
public class DefaultSliderModelAndRenderer<T> extends AbstractSliderModelAndRenderer<T> {

  public DefaultSliderModelAndRenderer() {
    super(true);
  }

  @Override
  protected void render(T element, JSlider slider, HTMLElement dataList, HTMLElement noDataList, HTMLElement option) {
    option.setAttribute("label", element.toString());
  }
}

File: transpile/javascript/swing/MnR/DefaultSliderModelAndRenderer.js
/**
 * The default implementation of the AbstractSliderModelAndRenderer
 *
 * @author gianpiero.diblasi
 * @param <T> The type
 */
class DefaultSliderModelAndRenderer extends AbstractSliderModelAndRenderer {

  /**
   * Creates the object
   */
  constructor() {
    super(true);
  }

   render(element, slider) {
    return element.toString();
  }
}
